cmd/init.go

```diff
@@ -892,60 +893,78 @@ func extractTarGz(gzipStream io.Reader) {
-			outFile, err := os.Create(header.Name)
-			if err != nil {
-				log.Println("extractTarGz: Create() failed: %s", err.Error())
-			}
-			if _, err := io.Copy(outFile, tarReader); err != nil {
-				log.Println("extractTarGz: Copy() failed: %s", err.Error())
+		p, _ := filepath.Abs(header.Name)
+		if !strings.Contains(p, "..") {
+		
+			switch header.Typeflag {
+				case tar.TypeDir:
+					if err := os.Mkdir(header.Name, 0755); err != nil {
+						log.Println("extractTarGz: Mkdir() failed: %s", err.Error())
+					}
+				case tar.TypeReg:
+					outFile, err := os.Create(header.Name)
+					if err != nil {
+						log.Println("extractTarGz: Create() failed: %s", err.Error())
+					}
+					if _, err := io.Copy(outFile, tarReader); err != nil {
+						log.Println("extractTarGz: Copy() failed: %s", err.Error())
+					}
+					outFile.Close()
+
+				default:
+					log.Println(
+						"extractTarGz: uknown type: %s in %s",
+						header.Typeflag,
+						header.Name)
 			}
-			outFile.Close()
-
-		default:
-			log.Println(
-				"extractTarGz: uknown type: %s in %s",
-				header.Typeflag,
-				header.Name)
 		}
 
 	}
 }
 
 func createSoftServe(kubeconfigPath string) {
+	createSoftServeFlag := viper.GetBool("create.softserve.create")	
+	if createSoftServeFlag != true {
+		log.Println("Executing createSoftServe")
+		if dryrunMode {
+			log.Printf("[#99] Dry-run mode, createSoftServe skipped.")
+			return
+		}
+		toolsDir := fmt.Sprintf("%s/.kubefirst/tools", home)
 
-	toolsDir := fmt.Sprintf("%s/.kubefirst/tools", home)
+		err := os.Mkdir(toolsDir, 0777)
+		if err != nil {
+			log.Println("error creating directory %s", toolsDir, err)
+		}
 
-	err := os.Mkdir(toolsDir, 0777)
-	if err != nil {
-		log.Println("error creating directory %s", toolsDir, err)
-	}
+		// create soft-serve stateful set
+		softServePath := fmt.Sprintf("%s/.kubefirst/gitops/components/soft-serve/manifests.yaml", home)
+		softServeApplyOut, softServeApplyErr,errSoftServeApply := execShellReturnStrings(kubectlClientPath, "--kubeconfig", kubeconfigPath, "-n", "soft-serve", "apply", "-f", softServePath, "--wait")
+		log.Printf("Result:\n\t%s\n\t%s\n",softServeApplyOut,softServeApplyErr)	
+		if errSoftServeApply != nil {
+			log.Println("failed to call kubectlCreateSoftServeCmd.Run(): %v", err)
+		}	
 
-	// create soft-serve stateful set
-	softServePath := fmt.Sprintf("%s/.kubefirst/gitops/components/soft-serve/manifests.yaml", home)
-	kubectlCreateSoftServeCmd := exec.Command(kubectlClientPath, "--kubeconfig", kubeconfigPath, "-n", "soft-serve", "apply", "-f", softServePath, "--wait")
-	kubectlCreateSoftServeCmd.Stdout = os.Stdout
-	kubectlCreateSoftServeCmd.Stderr = os.Stderr
-	err = kubectlCreateSoftServeCmd.Run()
-	if err != nil {
-		log.Println("failed to call kubectlCreateSoftServeCmd.Run(): %v", err)
+		viper.Set("create.softserve.create", true)
+		viper.WriteConfig()
+		log.Println("waiting for soft-serve installation to complete...")
+		time.Sleep(60 * time.Second)
```

Comment:
it could be a good idea to add a `todo:` to remember to update this blocker later on